File: solve3099.js
var sumOfTheDigitsOfHarshadNumber = function(x) {
    let a=String(x)
    let arr=a.split("")
    arr=arr.map((x)=>Number(x)).reduce((acc,val)=>acc+val,0)
    if(x%arr==0){
      return arr
    }else{
      return -1
    }
  };

File: solve3099.py
from functools import reduce
class Solution:
    def sumOfTheDigitsOfHarshadNumber(self, x: int) -> int:
        a=str(x)
        b=list(a)
        c=map(lambda x:int(x),b)
        c=reduce(lambda acc,val:acc+val,c,0)
        if x%c==0:
            return c
        else :
            return -1    
